### PR TITLE
Modified BufferPool.java

### PR DESCRIPTION
**Pull request by Rithesh17**
I have added codes for a few more methods

### CODE REVIEW

**Review by Ram-Aditya — APPROVED** [review]
All Good.

**Review by Priiyam — APPROVED** [review]

**Comment by sbshah97** [comment]
Why has this Pull Request not been merged?

**Review by manishsbhoopalam — APPROVED** [review]

**Comment by Ram-Aditya** [comment]
@salman-bhai We thought we needed to wait for your approval as we had requested a review from your side.
Sorry about that!We `ll go ahead with the merge now then.

**Pull request merged**
